ddtrace/contrib/anthropic/patch.py

```diff
@@ -0,0 +1,206 @@
+
+
+def get_version():
+    # type: () -> str
+    return getattr(anthropic, "__version__", "")
+
+
+config._add(
+    "anthropic",
+    {
+        "span_prompt_completion_sample_rate": float(os.getenv("DD_ANTHROPIC_SPAN_PROMPT_COMPLETION_SAMPLE_RATE", 1.0)),
+        "span_char_limit": int(os.getenv("DD_ANTHROPIC_SPAN_CHAR_LIMIT", 128)),
+    },
+)
+
+
+@with_traced_module
+def traced_chat_model_generate(anthropic, pin, func, instance, args, kwargs):
+    chat_messages = get_argument_value(args, kwargs, 0, "messages")
+    integration = anthropic._datadog_integration
+
+    operation_name = func.__name__
+
+    span = integration.trace(
+        pin,
+        "%s.%s" % (instance.__class__.__name__, operation_name),
+        submit_to_llmobs=True,
+        interface_type="chat_model",
+        provider="anthropic",
+        model=kwargs.get("model", ""),
+        api_key=_extract_api_key(instance),
+    )
+
+    chat_completions = None
+    try:
+        for message_idx, message in enumerate(chat_messages):
+            if not isinstance(message, dict):
+                continue
+            if isinstance(message.get("content", None), str):
+                if integration.is_pc_sampled_span(span):
+                    span.set_tag_str(
+                        "anthropic.request.messages.%d.content.0.text" % (message_idx),
+                        integration.trunc(message.get("content", "")),
+                    )
+                span.set_tag_str(
+                    "anthropic.request.messages.%d.content.0.type" % (message_idx),
+                    "text",
+                )
+            elif isinstance(message.get("content", None), list):
+                for block_idx, block in enumerate(message.get("content", [])):
+                    if integration.is_pc_sampled_span(span):
+                        if block.get("type", None) == "text":
+                            span.set_tag_str(
+                                "anthropic.request.messages.%d.content.%d.text" % (message_idx, block_idx),
+                                integration.trunc(str(block.get("text", ""))),
+                            )
+                        elif block.get("type", None) == "image":
+                            span.set_tag_str(
+                                "anthropic.request.messages.%d.content.%d.text" % (message_idx, block_idx),
+                                "([IMAGE DETECTED])",
+                            )
+
+                    span.set_tag_str(
+                        "anthropic.request.messages.%d.content.%d.type" % (message_idx, block_idx),
+                        block.get("type", "text"),
+                    )
+            span.set_tag_str(
+                "anthropic.request.messages.%d.role" % (message_idx),
+                message.get("role", ""),
+            )
+        params_to_tag = {k: v for k, v in kwargs.items() if k != "messages"}
+        span.set_tag_str("anthropic.request.parameters", json.dumps(params_to_tag))
```

Comment:
Just curious, how do tool calls get captured by this? I know tools can get very long so this may be excessive to tag and I would prefer avoiding tagging this parameter as well.

Reply:
I would also prefer avoiding tagging `models` again since we already tag it as a base level tag, can we keep a small blacklist of kwarg names to not tag as parameters?

Reply:
Ill add a test for tools, also yes ill add a blacklist

Reply:
I've added a tools test, and added tools to the blacklist so it doesn't capture the param, same with model.